azure/scope/managedcontrolplane.go

```diff
@@ -625,6 +625,13 @@ func (s *ManagedControlPlaneScope) SetKubeConfigData(kubeConfigData []byte) {
 	s.kubeConfigData = kubeConfigData
 }
 
+// SetKubeletIdentity sets the ID of the user-assigned identity for kubelet if not already set.
+func (s *ManagedControlPlaneScope) SetKubeletIdentity(id string) {
+	if s.ControlPlane.Spec.KubeletUserAssignedIdentity == "" {
```

Comment:
Do we expect AKS would ever mutate the identity? Because if so, it _may_ cause a bug in the future where the persisted identity and the actual identity drift. It might be more future proof to always set it to what the actual value is. It shouldn't overwrite the user's value since we're setting it from the result AKS cluster which was built from the desired spec.

Reply:
Makes sense, removed.